git-workflow.md

```diff
@@ -0,0 +1,101 @@
+# Pacific Development `git` workflow
+
+## Single source of truth
+
+The `main` branch of [this](https://github.com/PacificDevelopment/JobSite) remote repository will serve as our **single source of truth**. It will represent the latest version of production-ready code.
+
+## Branch Protection
+`main` branch has special protection rules.
+1. To merge code into `main` you must create a pull request.
+2. The pull request must be approved by at least 1 other person.
+3. No one can push directly to main except the team captains.
+4. In the future Continous integration will also place a status check preventing anyone from pushing directly to main.
+
+
+## Cloning the repository
+
+Your first step will be cloning our remote repository. Navigate to the appropriate folder on your local machine and execute the following command:
+
+```
+git clone https://github.com/PacificDevelopment/JobSite.git <desired_folder_name>
+```
+
+Then change into your newly created directory:
+
+```
+cd <desired_folder_name>
+```
+
+Confirm that you are configured with the correct remote repository:
+
+```
+git remote -v
+```
+
+You should see the following:
+
+```
+origin	https://github.com/PacificDevelopment/JobSite.git (fetch)
+origin	https://github.com/PacificDevelopment/JobSite.git (push)
+```
+
+## Working on a feature
+
+> No work should be done on the `main` branch.
+
+All work should be done on a **feature branch**; that is, an existing or newly created branch designated to the feature on which you are working.
+
+Resoruce for more about feature branch workflow: https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow
+
+### Create a new branch
+
+From the `main` branch, create and checkout a new branch:
+
+```
+git checkout -b <new_feature_branch_name>
+```
+
+### Implementing changes to your feature
+
+Implement the changes to your feature's code, making frequent, small commits as follows:
+
+```
+git add .
+git commit -m "<your_commit_message>"
+```
+
+Use descriptive commit messages in the present tense; for example:
+
+```
+git commit -m "implement click event handler on checkout button"
+```
+Resource for writing good commit messages: https://cbea.ms/git-commit/
+
```

Comment:
Including outside resources === very helpful!!

Reply:
Happy to help!